src/HttpClientPoolItem.php

```diff
@@ -16,7 +16,7 @@
  *
  * @author Joel Wurtz <joel.wurtz@gmail.com>
  */
-class HttpClientPoolItem implements HttpClient, HttpAsyncClient
+final class HttpClientPoolItem implements HttpClient, HttpAsyncClient
```

Comment:
Same.. this client also does not have an abstraction layer for all of it's methods.

Reply:
this should not be used outside of the library itself. but using annotations now so that we can still test it.